gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +969,920 @@ public void testListDnsRecords() {
   }
 
   @Test
-  public void testListZoneBatch() {
-    DNS.create(ZONE1);
-    DNS.create(ZONE_EMPTY_DESCRIPTION);
+  public void testListZonesBatch() {
     try {
       DnsBatch batch = DNS.batch();
-      DnsBatchResult<Page<Zone>> batchResult = batch.listZones();
+      DnsBatchResult<Page<Zone>> result = batch.listZones();
+      batch.submit();
+      List<Zone> zones = filter(result.get().iterateAll());
+      assertEquals(0, zones.size());
+      // some zones exists
+      Zone firstZone = DNS.create(ZONE1);
+      batch = DNS.batch();
+      result = batch.listZones();
       batch.submit();
-      assertTrue(batchResult.completed());
-      Iterator<Zone> iteratorBatch = batchResult.get().iterateAll();
-      Iterator<Zone> iteratorList = DNS.listZones().iterateAll();
-      assertEquals(Iterators.size(iteratorList), Iterators.size(iteratorBatch));
-      while (iteratorBatch.hasNext()) {
-        assertTrue(Iterators.contains(iteratorList, iteratorBatch.next()));
+      zones = filter(result.get().iterateAll());
+      assertEquals(firstZone, zones.get(0));
+      assertEquals(1, zones.size());
```

Comment:
Maybe assert size before getting the first element? To get a nice failure rather than an error